CHANGELOG.md

```diff
@@ -311,7 +311,15 @@ Pragmas and options
 
 * Option `--count-clusters` is now on by default when Agda was built with ICU support (Cabal flag `enable-cluster-counting`).
 
-* Now pragma `INJECTIVE_FOR_INFERENCE`, which treats functions as injective for inferring implicit arguments.
+* New pragma `INJECTIVE_FOR_INFERENCE`, which treats functions as injective for inferring implicit arguments. For example:
+  ```agda
+  reverse-≡ : {l l' : List A} → reverse l ≡ reverse l' → reverse l ≡ reverse l'
+  reverse-≡ h = h
+
+  []≡[] : {l l' : List A} → [] ≡ []
+  []≡[] = reverse-≡ (refl {x = reverse []})
+  ```
+  does not work since Agda won't solve `l` and `l'` for `[]`, even though it knows `reverse l = reverse []`. If `reverse` is marked as injective with `{-# INJECTIVE_FOR_INFERENCE reverse #-}` this example will work.
```

Comment:
Thanks for the example.  

But mmh...
To the outsider this looks still very artificial, because the first function is just the identity and the second looks very pointless, with implicit arguments `l l'` that can never be found automatically and a statement that has just `refl` as its shortest proof.

Reply:
Yeah, the issue is that as soon as you have a function that Agda can detect as being injective in there it'll infer the arguments - and Agda is quite good at that! The implicit arguments to the second functions were of course just a leftover that I forgot about.

Here's another artificial one, but I think it's obvious how to replace it with your favourite property:
```agda
  P : List A → List A → Set
  P _ _ = Unit

  {-# INJECTIVE_FOR_INFERENCE reverse #-}

  reverse-≡-P : {l l' : List A} → reverse l ≡ reverse l' → P l l'
  reverse-≡-P _ = tt

  P-[]-[] : P [] []
  P-[]-[] = reverse-≡-P {l = []} refl
```

Reply:
I could construct something more interesting with permutations or membership, but that'll blow up the size of the example.

Reply:
Why not simply:
```agda
reverse : List A -> List A
reverse = ...

{-# INJECTIVE_FOR_INFERENCE reverse #-}

reverse-injective : {l l' : List A} → reverse l ≡ reverse l' → l ≡ l'
reverse-injective refl = refl
```